src/clj/game/cards/hardware.clj

```diff
@@ -398,12 +398,21 @@
                   :effect (req (add-counter state :runner card :virus -1)
                                (add-counter state :runner target :virus 1))}]
      {:events {:runner-turn-begins ability
-               :runner-trash {:req (req (= (:side target) "Corp"))
-                              :optional
-                              {:prompt "Gain a virus counter on Friday Chip?"
-                               :yes-ability
-                               {:effect (effect (add-counter :runner card :virus 1)
-                                                (system-msg :runner (str "places 1 virus counter on Friday Chip")))}}}}})
+               :runner-trash {:delayed-completion true
+                              :req (req (some #(card-is? % :side :corp) targets))
+                              :effect (req (let [amt-trashed (count (filter #(card-is? % :side :corp) targets))
+                                                 sing-ab {:optional {:prompt "Place a virus counter on Friday Chip?"
+                                                                     :yes-ability {:effect (effect (add-counter :runner card :virus 1))
+                                                                                   :msg "place 1 virus counter on Friday Chip"}}}
+                                                 mult-ab {:prompt "Place virus counters on Friday Chip?"
+                                                          :choices {:number (req amt-trashed)
+                                                                    :default (req amt-trashed)}
+                                                          :msg (msg "place " (quantify target "virus counter") " on Friday Chip")
+                                                          :effect (effect (add-counter :runner card :virus target))}
+                                                 ab (if (> amt-trashed 1) mult-ab sing-ab)]
+                                             (resolve-ability state side eid ab card targets)))}
```

Comment:
I prefer you to use `(continue-ability` here without the `eid`. This is ultimately what `continue-ability` does, but I think it's more readable than having to see if `eid` is passed correctly by hand.